### PR TITLE
A simple fix for occasional errors when using sffs

### LINKED ISSUE #2485: sffs crashes
Using `sffs` may produce the following error (see the minimal reproducible example below)

> Error in resample(lrn, mCT, outer, extract = getFeatSelResult, measures = list(mlr::auc,  : 
  Assertion on 'x' failed: Must be of type 'integerish', not 'list'.

The error does not appear when using, for example, 'sbs'  feature selection. This issue was initially raised here:  <https://stackoverflow.com/questions/53263251/how-to-jointly-use-makefeatselwrapper-and-resample-function-in-mlr/53283284#53283284> 

Minimal reproducible example:
```r
library(mlbench)
library(mlr)

p <- mlbench.waveform(1000) 
dataset <- as.data.frame(p) 
dataset  <- droplevels(subset(dataset, classes != 3))

mCT <- makeClassifTask(data = dataset, target = "classes")
ctrl = makeFeatSelControlSequential(method = "sffs", maxit = NA,alpha = 0.001)
mL <- makeLearner("classif.logreg", predict.type = "prob")
inner = makeResampleDesc("Holdout",stratify = TRUE)
lrn = makeFeatSelWrapper(mL, resampling = inner, control = ctrl)
outer = makeResampleDesc("CV", iters = 10,stratify = TRUE)
resample(lrn, mCT, outer, extract = getFeatSelResult,measures = list(mlr::auc,mlr::acc,mlr::brier), models=TRUE)  
```

`sessionInfo()`:
 
```
R version 3.5.1 (2018-07-02)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.5 LTS

Matrix products: default
BLAS: /usr/lib/libblas/libblas.so.3.6.0
LAPACK: /usr/lib/lapack/liblapack.so.3.6.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=es_ES.UTF-8        LC_COLLATE=en_US.UTF-8     LC_MONETARY=es_ES.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=es_ES.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=es_ES.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] mlr_2.13.9000     ParamHelpers_1.11 mlbench_2.1-1    

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.0        magrittr_1.5      bindr_0.1.1       splines_3.5.1     tidyselect_0.2.4  munsell_0.5.0     lattice_0.20-35   colorspace_1.3-2  R6_2.3.0         
[10] rlang_0.3.0.1     fastmatch_1.1-0   plyr_1.8.4        dplyr_0.7.6       tools_3.5.1       parallel_3.5.1    grid_3.5.1        checkmate_1.8.5   data.table_1.11.8
[19] gtable_0.2.0      survival_2.42-3   yaml_2.2.0        lazyeval_0.2.1    assertthat_0.2.0  tibble_1.4.2      Matrix_1.2-14     bindrcpp_0.2.2    purrr_0.2.5      
[28] ggplot2_3.1.0     glue_1.2.0        parallelMap_1.4   stringi_1.2.4     compiler_3.5.1    pillar_1.2.3      BBmisc_1.11       scales_1.0.0      backports_1.1.2  
[37] XML_3.98-1.16     pkgconfig_2.0.1  
```

### PR DESCRIPTION
**Pull request by bmihaljevic**
This is one-line fix for https://github.com/mlr-org/mlr/issues/2485. The issue is related to occasional crashing when using `sffs` feature selection.

### CODE REVIEW

**Comment by pat-s** [comment]
@bmihaljevic Thanks for contributing. 

- Please edit your title so that it has a proper meaning
- Reference the issue in the PR description
- Do not simply copy/paste the PR template

**Comment by berndbischl** [comment]
More importantly : doesn't this need a test?

**Comment by bmihaljevic** [comment]
I have just added a simple test that ensures that the minimal example from https://github.com/mlr-org/mlr/issues/2485 no longer raises an error.

File: tests/testthat/test_featsel_selectFeaturesSequential.R [hunk 1]
```diff
@@ -0,0 +1,18 @@
+context("selectFeaturesSequential")
+
+# This used to cause an error. Now ensure it does not.
+test_that("no crash with sffs", {
+  p <- mlbench::mlbench.waveform(1000)
```
Comment (larskotthoff) [inline tests/testthat/test_featsel_selectFeaturesSequential.R]: Can you use `=` for assignment please?

File: tests/testthat/test_featsel_selectFeaturesSequential.R [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  dataset <- as.data.frame(p)
+  dataset  <- droplevels(subset(dataset, classes != 3))
+
+  mCT <- makeClassifTask(data = dataset, target = "classes")
+  ctrl = makeFeatSelControlSequential(method = "sffs", maxit = NA,alpha = 0.001)
```
Comment (larskotthoff) [inline tests/testthat/test_featsel_selectFeaturesSequential.R]: Please add a space after comma.

File: tests/testthat/test_featsel_selectFeaturesSequential.R [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  mL <- makeLearner("classif.logreg", predict.type = "prob")
+  inner = makeResampleDesc("Holdout",stratify = TRUE)
+  lrn = makeFeatSelWrapper(mL, resampling = inner, control = ctrl)
+  outer = makeResampleDesc("CV", iters = 10,stratify = TRUE)
```
Comment (larskotthoff) [inline tests/testthat/test_featsel_selectFeaturesSequential.R]: Would that work with a smaller number of folds or a simple holdout? We try to keep the tests as quick as possible.

**Comment by bmihaljevic** [comment]
Hi, 
I think I have addressed all suggested changes. Please let me know if I am missing something that is pending on my side.
Thanks,
Bojan

**Comment by pat-s** [comment]
@larskotthoff 	Should we merge this? We now have a test. I don't know if the proposed change could break anything else but as long as the build is passing...

We should also add a NEWS entry.

**Comment by larskotthoff** [comment]
Did the travis build for this pass? It seemed to not want to start earlier.

**Comment by pat-s** [comment]
Yours passed and I just added an NEWS entry, therefore I did not wait. Master is green.

**Comment by larskotthoff** [comment]
Ok, thanks!

**Pull request merged**